tests/test_load.py

```diff
@@ -1,33 +1,225 @@
+    assert os.path.exists("./test_staging_dir")
+    os.rmdir("./test_staging_dir")
+
+
+def test_delete_temp_location():
+    os.mkdir("./test_staging_dir")
+    load.delete_temp_location(staging_path="./test_staging_dir")
+    assert not os.path.exists("./test_staging_dir")
+
+
+def test_remove_non_values():
+    example_dict = {
+        "a": {"b": "c"},  # is dictionary
+        "d": {"e": "f", "g": None},  # is dictinoary with nested none
+        "h": ["i", "j", "k"],  # is list
+        "l": None,  # is none
+        "m": "n",  # is anything else
+    }
+    cleaned_dict = load.remove_non_values(example_dict)
+    assert isinstance(cleaned_dict, dict)
+    assert cleaned_dict["a"] == example_dict["a"]
+    assert cleaned_dict["d"] == {"e": "f"}
+    assert cleaned_dict["h"] == example_dict["h"]
+    assert cleaned_dict["m"] == example_dict["m"]
+    assert cleaned_dict.get("l") is None
+
+
+class TestLoad:
+    @pytest.fixture(scope="function", autouse=True)
+    def setup_method(self, syn):
+        self.patch_syn_login = patch.object(
+            utils, "_login_to_synapse", return_value=syn
+        ).start()
+        self.patch_syn_store = patch.object(
+            syn,
+            "store",
+            return_value=File(
+                "fake/path/to/fake/file",
+                parent="syn1111113",
+                id="syn1111114",
+                versionNumber=1,
+            ),
+        ).start()
+
+    def teardown_method(self):
+        mock.patch.stopall()
+
+    def test_load_syn_is_none(self):
+        test_tuple = load.load(
+            file_path="fake/path/to/fake/file",
+            provenance=["syn1111111", "syn1111112"],
+            destination="syn1111113",
+            syn=None,
+        )
+        self.patch_syn_login.assert_called_once()
+        assert test_tuple == ("syn1111114", 1)
+
+    def test_load_syn_is_not_none(self, syn):
+        test_tuple = load.load(
+            file_path="fake/path/to/fake/file",
+            provenance=["syn1111111", "syn1111112"],
+            destination="syn1111113",
+            syn=syn,
+        )
+        self.patch_syn_login.assert_not_called()
+        assert test_tuple == ("syn1111114", 1)
+
+    def test_load_activity_fails_ValueError(self):
+        captured_output = StringIO()
+        sys.stdout = captured_output
+        test_tuple = load.load(
+            file_path="fake/path/to/fake/file",
+            provenance=["not a syn id", "also not a syn id"],
+            destination="syn1111113",
+            syn=None,
+        )
+        self.patch_syn_login.assert_called_once()
+        assert "one or more invalid syn ids" in captured_output.getvalue()
+        self.patch_syn_store.assert_not_called()
+        assert test_tuple is None
+
+    def test_load_syn_store_fails_OSError(self):
+        self.patch_syn_store.side_effect = OSError  # can't induce failure in mocked object with arguments, so do manually
```

Comment:
Learned how to make mocked objects throw specific errors - handy because you can't make them throw errors by passing bad arguments to them.